packages/visx-tooltip/src/hooks/useTooltipInPortal.tsx

```diff
@@ -40,31 +43,44 @@ export default function useTooltipInPortal({
   const TooltipInPortal = useMemo(
     () =>
       ({
-        left: containerLeft = 0,
-        top: containerTop = 0,
+        left: tooltipLeft = 0,
+        top: tooltipTop = 0,
         detectBounds: detectBoundsProp, // allow override at component-level
+        portalContainer: portalContainerProp, // allow override at component-level
         zIndex: zIndexProp, // allow override at the component-level
         ...tooltipProps
       }: TooltipInPortalProps) => {
         const detectBounds = detectBoundsProp == null ? detectBoundsOption : detectBoundsProp;
+        const portalContainer =
+          portalContainerProp == null ? portalContainerOption : portalContainerProp;
+        const portalContainerRect = portalContainer?.getBoundingClientRect();
         const zIndex = zIndexProp == null ? zIndexOption : zIndexProp;
         const TooltipComponent = detectBounds ? TooltipWithBounds : Tooltip;
         // convert container coordinates to page coordinates
-        const portalLeft = containerLeft + (containerBounds.left || 0) + window.scrollX;
-        const portalTop = containerTop + (containerBounds.top || 0) + window.scrollY;
+        const portalLeft = portalContainer
+          ? tooltipLeft - (portalContainerRect?.left || 0) + (containerBounds.left || 0)
+          : tooltipLeft + (containerBounds.left || 0) + window.scrollX;
+        const portalTop = portalContainer
+          ? tooltipTop - (portalContainerRect?.top || 0) + (containerBounds.top || 0)
+          : tooltipTop + (containerBounds.top || 0) + window.scrollY;
+
+        const additionalTooltipProps = portalContainer ? { parentRect: containerBounds } : {};
 
         return (
-          <Portal zIndex={zIndex}>
-            <TooltipComponent left={portalLeft} top={portalTop} {...tooltipProps} />
+          <Portal container={portalContainer} zIndex={zIndex}>
+            <TooltipComponent
+              left={portalLeft}
+              top={portalTop}
+              {...tooltipProps}
+              {...additionalTooltipProps}
+            />
           </Portal>
         );
       },
-    [detectBoundsOption, zIndexOption, containerBounds.left, containerBounds.top],
+    [containerBounds, detectBoundsOption, portalContainerOption, zIndexOption],
```

Comment:
I think it's ideal to have this reference `containerBounds.left/top` explicitly since the object `containerBounds` might not be referentially equal across renders.

Reply:
Totally fair, I assumed that would never be the case based on the type of `containerBounds` being `RectReadOnly` where `left` and `top` are readonly properties, but better safe than sorry. 👍

Reply:
Ha, actually, eslint is yelling at me saying that `containerBounds.left` and `containerBounds.top` are unnecessary dependencies when I add them. 🙈 Do you think there's actually any reasonable change for the `readonly` modifier on `left` and `top` to be violated on this object?